### PR TITLE
Renamed TagWillBeSaved => Saving. Deprecated the old event.

### PR DESCRIPTION
**Pull request by iPurpl3x**
As discussed in #86, here is a PR for the Saving event that was previously named TagWillBeSaved but should be renamed to match the general convention.
Is the `@deprecated beta 14, removed beta 15.` comment that I made right @askvortsov1 ?

### CODE REVIEW

File: src/Event/TagWillBeSaved.php [hunk 1]
```diff
@@ -12,6 +12,9 @@
 use Flarum\Tags\Tag;
 use Flarum\User\User;
 
+/**
+ * @deprecated beta 14, removed beta 15.
+ */
```
Comment (franzliedke) [inline src/Event/TagWillBeSaved.php:17]: If we keep this, we would need to keep raising it as well.

Alternative (and my preference): Just remove the event. :wink:

**Review by franzliedke — CHANGES_REQUESTED** [review]

File: src/Event/TagWillBeSaved.php (hunk 1 above)
Comment (franzliedke) [reply src/Event/TagWillBeSaved.php:17]: Pretty much what Clark said, sorry.

Comment (franzliedke) [reply src/Event/TagWillBeSaved.php:17]: Option 1 was actually requested by @askvortsov1 in https://github.com/flarum/tags/pull/86#issuecomment-646032898 as well.

**Comment by luceos** [comment]
Looking at the discussion this is what still has to be done:

- [ ] Keep firing the deprecated event
- [ ] Create a new issue to remove that deprecated event and assign it to the next release/milestone 

@iPurpl3x could you tackle this so that we can ship it with the upcoming release?

@clarkwinkelmann let's create an issue for that, I think we need to make this a convention anyhow. We can then handle this in a new PR across all core extensions.

**Pull request merged**